### PR TITLE
add skinny deck

### PR DESCRIPTION
**Pull request by tsankuanglee**
The current `library/assets/Deck/4.json` uses external sources for assets. This PR:
* downloads the source and renames the files, using the same filename convention of `library/assets/Deck/0.json` (default deck)
* `4.json` file provides the same fields as seen in `0.json` to support various sorting keys.
* add two more colors of jokers, to match the 4-color convention for other ranks.
* updated attribution to be more specific.

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: https://test.virtualtabletop.io/PR-1619/pr-test (or any other room on that server)

After merging, a backup will be available at `https://beta.virtualtabletop.io/editor/PR1619-pr-test`.

**Comment by tsankuanglee** [comment]
Reviewers, the shuffle button doesn't work -- the [routine](https://github.com/ArnoldSmith86/virtualtabletop/blob/fe35d35e070d3a3bd51dbddf8bda4b71066fb8d0/library/assets/Decks/4.json#L594-L609) is automatically migrated to:

```
  "clickRoutine": [
    "var internal_computeMigration_existingVariable0 = '\\u0024\\u007bPROPERTY\\u0020parent\\u007d' // This was added by the automatic file migration because the new expression syntax does not support escaping variable expressions.",
    {
      "func": "RECALL",
      "holder": "${internal_computeMigration_existingVariable0}"
    },
    "var internal_computeMigration_existingVariable1 = '\\u0024\\u007bPROPERTY\\u0020parent\\u007d' // This was added by the automatic file migration because the new expression syntax does not support escaping variable expressions.",
    {
      "func": "FLIP",
      "holder": "${internal_computeMigration_existingVariable1}",
      "face": 0
    },
    "var internal_computeMigration_existingVariable2 = '\\u0024\\u007bPROPERTY\\u0020parent\\u007d' // This was added by the automatic file migration because the new expression syntax does not support escaping variable expressions.",
    {
      "func": "SHUFFLE",
      "holder": "${internal_computeMigration_existingVariable2}"
    }
  ]
```
Ideas?

**Comment by ArnoldSmith86** [comment]
- Those images do not match https://opengameart.org/content/casino-pack - that's a different deck in another variant.
- The metadata in 0-5.json is expected to be the same. Most of the fields are _state_ level so they apply to all variants. They are simply saved in all variant JSON files because there is no overarching state JSON file. Exceptions are `players`, `language`, `variant`, `variantImage` and special stuff like `link`, `plStateID` and `plVariantID`. This means that your changed `attribution` and `image` are not visible anywhere unless you change it in all variants. The `attribution` is a bit messy in the Decks assets because it used to be a per variant metadata field but now it is a per state metadata field and the migration just concatenated the attribution for all the variants. This could use a cleanup to make it more obvious which attribution belongs to which deck.
- The stars in the joker images could use centering. Looks like they are too far left.

**Comment by robartsd** [comment]
> The `attribution` is a bit messy in the Decks assets because it used to be a per variant metadata field but now it is a per state metadata field and the migration just concatenated the attribution for all the variants.

I think this is generally a problem in the Assets section. Would be nice to be able to have attribution for the overall state collection as well as attribution per variant. Perhaps we need to add a place specifically to save state collection level meta data.

**Comment by ArnoldSmith86** [comment]
The Assets section will dissolve soon anyway.

**Comment by RaphaelAlvez** [comment]
Add a variant to the assets "game" with a deck that uses your files?

**Comment by tsankuanglee** [comment]
> Those images do not match https://opengameart.org/content/casino-pack - that's a different deck in another variant.

Thanks for catching that. @ArnoldSmith86 . This is then a bit tricky since [the source](https://github.com/pwinckles/pcio) doesn't explicitly state the license/terms. I had assumed that since we included that in the assets section, we must've cleared the rights. But now that I think more about it, it might be a grey area -- technically github hosts it, not vtt servers; maybe that's why we used hyperlinks instead of local files?

Can somebody help double verify the right of these files? Or maybe we can just ask Peter Winkles (the repo owner) himself?

**Comment by 96LawDawg** [comment]
> > Those images do not match https://opengameart.org/content/casino-pack - that's a different deck in another variant.
> 
> Thanks for catching that. @ArnoldSmith86 . This is then a bit tricky since [the source](https://github.com/pwinckles/pcio) doesn't explicitly state the license/terms. I had assumed that since we included that in the assets section, we must've cleared the rights. But now that I think more about it, it might be a grey area -- technically github hosts it, not vtt servers; maybe that's why we used hyperlinks instead of local files?
> 
> Can somebody help double verify the right of these files? Or maybe we can just ask Peter Winkles (the repo owner) himself?

The absence of a license means no license.  However, if you are specifically talking about the skinny deck, that was created by an anonymous person and given to VTT and released into the public domain:  "The cards in this deck were created by an anonymous person who gave the work to virtualtabletop.io. Virtualtabletop.io releases all of this to the Public Domain under CC0."

**Comment by tsankuanglee** [comment]
> "The cards in this deck were created by an anonymous person who gave the work to virtualtabletop.io. Virtualtabletop.io releases all of this to the Public Domain under CC0."

That's a good news. Wasn't sure which license text corresponds to which section. Now that it's cleared, I'll fix the notes.

**Comment by tsankuanglee** [comment]
> Add a variant to the assets "game" with a deck that uses your files?

@RaphaelAlvez what do you mena by assets "game"? Right now, I'm replacing the json of the Domino variant of the assets Decks. I thought that should suffice for users to use. No?

**Comment by tsankuanglee** [comment]
@ArnoldSmith86 
> * The stars in the joker images could use centering. Looks like they are too far left.

Good eyes! It turned out that there are many cards that were not centered perfectly. I re-aligned all of them now.

**Comment by tsankuanglee** [comment]
Ready for review now. Thanks all!

File: assets/cards-skinny/credit.md [hunk 1]
```diff
@@ -0,0 +1 @@
+The cards in this deck were created by an anonymous person who gave the work to virtualtabletop.io.  Virtualtabletop.io releases all of this to the Public Domain under CC0.
```
Comment (RaphaelAlvez) [inline assets/cards-skinny/credit.md:1]: Hum isn't there a source? Shouldn't we give credits to the username?

Also was it given directly to VTT or was it made public and we took it?
The "given to VTT" and "we are releasing it as CC0" makes me feel there is something wrong.

**Review by RaphaelAlvez — CHANGES_REQUESTED** [review]
Code review and tested. Approved appart from the comment I made.

I have one concerne about the attribution but after that's discussed I'll approve it

File: assets/cards-skinny/credit.md (hunk 1 above)
Comment (ArnoldSmith86) [reply assets/cards-skinny/credit.md:1]: That's unrelated to the PR though. The same text was in the attribution of the variant before.

**Comment by RaphaelAlvez** [comment]
> > Add a variant to the assets "game" with a deck that uses your files?
> 
> @RaphaelAlvez what do you mena by assets "game"? Right now, I'm replacing the json of the Domino variant of the assets Decks. I thought that should suffice for users to use. No?
> 

Please ignore that comment. The hight number of file changes was messing up with GitHub.

File: assets/cards-skinny/credit.md (hunk 1 above)
Comment (96LawDawg) [reply assets/cards-skinny/credit.md:1]: > Hum isn't there a source? Shouldn't we give credits to the username?
> 
> Also was it given directly to VTT or was it made public and we took it? The "given to VTT" and "we are releasing it as CC0" makes me feel there is something wrong.

The person who made this deck did not want attribution by name.  They made it, gave it to VTT to do with whatever.  I have the message traffic somewhere in Discord.

**Review by RaphaelAlvez — APPROVED** [review]
Approved

**Comment by 96LawDawg** [comment]
Just so that it is documented ...  The creator of the skinny deck was figimir.  Here is the link on Discord: https://discord.com/channels/770758631146782780/789103957938012171/797827238846005308.  And it was also mentioned in #102.

**Comment by 96LawDawg** [comment]
@tsankuanglee, I updated the attribution metadata in commit [bcf6517](https://github.com/ArnoldSmith86/virtualtabletop/pull/1619/commits/bcf6517157393e684164740cf1569bdf7c122b10), but those changes are not showing up in the test room because that is not the main variant.  Can you please make those changes to variant 0?

**Comment by ArnoldSmith86** [comment]
Like I said, the metadata should be the same in all variants except for variant specific properties.

**Comment by tsankuanglee** [comment]
@96LawDawg thanks for making the credit text more explicit. I've ported it to other variants.

**Review by 96LawDawg — APPROVED** [review]
Cards look good.  Attribution looks code.  Approved.

**Pull request merged**